odoo/addons/base/transpiler/sourcemap_generator.py

```diff
@@ -0,0 +1,168 @@
+from functools import lru_cache
+
+
+class SourceMapGenerator:
+    """
+    The SourceMapGenerator creates the sourcemap maps the asset bundle to the js files.
+
+    What is a sourcemap ? (https://developer.mozilla.org/en-US/docs/Tools/Debugger/How_to/Use_a_source_map)
+    In brief: a source map is what makes possible to debug your processed/compiled/minified code as your were debugging
+    the original, non-altered source code. It is a file that provides a mapping original <=> processed for the
+    browser to read.
+
+    This implementation of the SourceMapGenerator is a translation and adaptation of this implementation
+    in js https://github.com/mozilla/source-map. For performance purposes, we have removed all unnecessary
+    functions/steps for our use case. This simpler version does a line by line mapping, with the ability to
+    add offsets at the start and end of a file. (when we have to add comments on top a transpiled file by example).
+    """
+    def __init__(self, **kwargs):
+        self._file = kwargs.get("file")
+        self._sourceRoot = kwargs.get("sourceRoot")
```

Comment:
```suggestion
    def __init__(self, file=None, sourceRoot=None):
        self._file = file
        self._sourceRoot = sourceRoot
```

Reply:
Also `sourceRoot` should probably be renamed `source_root` at this point.

And we don't seem to use `file`, maybe remove it entirely?